src/targets/Logary.Targets.Stackdriver/Target_Stackdriver.fs

```diff
@@ -0,0 +1,173 @@
+    | LogLevel.Verbose -> LogSeverity.Debug
+
+// exposed to allow for easier testing
+let write (m : Message) : LogEntry =
+  let values = 
+    match Json.serialize m with
+    | Object values -> values
+    | otherwise -> failwithf "Expected Message to format to Object .., but was %A" otherwise
+  
+  let overrides = Map.add "name" (String <| string m.name) values
+  let json = Google.Protobuf.WellKnownTypes.Struct.Parser.ParseJson (Json.format <| Object overrides)
+  LogEntry(Severity = toSeverity m.level, JsonPayload = json)
+
+module internal Impl =
+
+  type State = {
+    logName : string
+    logger : LoggingServiceV2Client
+    resource : MonitoredResource
+  }
+
+  let mkLabels project resource =
+    match resource with
+    // check out https://cloud.google.com/logging/docs/api/v2/resource-list for the list of resources
+    // as well as required keys for each one
+    | ComputeInstance(zone, instance) -> [ "project_id", project
+                                           "zone", zone
+                                           "instanceId", instance ]|> dict
+    | Container(cluster, ns, instance, pod, container, zone) -> [ "project_id", project
+                                                                  "cluster_name", cluster
+                                                                  "namespace_id", ns
+                                                                  "instance_id", instance
+                                                                  "pod_id", pod
+                                                                  "container_name", container
+                                                                  "zone", zone ] |> dict
+  let resourceName = function
+  | ComputeInstance _ -> "gce_instance"
+  | Container _ -> "container"
+  
+  let mkMonitoredResource project resourceType =
+    let r = MonitoredResource(Type = resourceName resourceType)
+    r.Labels.Add(mkLabels project resourceType)
+    r
+
+  let writeRequests (conf : StackdriverConf) (state : State) (reqs : TargetMessage[]) =
+      let messages = ResizeArray<_>() // Updated by the loop.
+      let acks = ResizeArray<_>() // updated by the loop
+      
+      for m in reqs do
+        match m with
+        | Log (message, ack) ->
+          // Write and save the ack for later.
+          acks.Add ack
+          messages.Add <| write message
+        | Flush (ackCh, nack) -> ()
+
+      let writer = Job.Scheduler.isolate <| fun _ -> state.logger.WriteLogEntries(state.logName, state.resource, conf.labels, messages)   
+      let handler = Job.tryIn writer Job.result (Job.raises)
+      handler >>= fun res -> Job.conIgnore acks 
+  
+  let createState (conf : StackdriverConf) = 
+    { logger = LoggingServiceV2Client.Create()
+      resource = mkMonitoredResource conf.projectId conf.resource
+      logName = sprintf "projects/%s/logs/%s" conf.projectId conf.logId }
+
+  let dispose runtime state ackVar = 
```

Comment:
If there's nothing to do in here, we can probably move this single line of code into the main loop where you handle the shutdown signal?